src/quic/node_quic_util-inl.h

```diff
@@ -323,23 +330,29 @@ template <typename T>
 StatsBase<T>::StatsBase(
     Environment* env,
     v8::Local<v8::Object> wrap,
-    int options)
-    : stats_buffer_(
-          env->isolate(),
-          sizeof(typename T::Stats) / sizeof(uint64_t),
-          reinterpret_cast<uint64_t*>(&stats_)) {
+    int options) {
   static constexpr uint64_t kMax = std::numeric_limits<int64_t>::max();
-  stats_.created_at = uv_hrtime();
-
-  // TODO(@jasnell): The follow are checks instead of handling
-  // the error. Before this code moves out of experimental,
-  // these should be change to properly handle the error.
 
-  wrap->DefineOwnProperty(
+  // Create the backing store for the statistics
+  size_t size = sizeof(typename T::Stats);
+  size_t count = size / sizeof(uint64_t);
+  stats_store_ = v8::ArrayBuffer::NewBackingStore(env->isolate(), size);
+  stats_ = new (stats_store_->Data()) typename T::Stats;
```

Comment:
Not sure if it makes a difference right now, but generally there should be a call to `~typename T::Stats()` when placement new is being used (probably from ~StatsBase).

You should also be able to do something like `typedef typename T::Stats Stats;` in the class definition, so that you don’t always have to spell out `typename T::Stats`.